tests/soldier.test.ts

```diff
@@ -32,63 +27,80 @@ describe('/soldiers routes', () => {
 
   describe('POST /soldiers', () => {
     test('Should return code 201 and object if all info passed correctly', async () => {
+      const soldier = createTestSoldier(0);
       const res = await request(server)
         .post(`/soldiers`)
-        .send(lodash.omit(testSoldiers[0], 'duties'));
+        .send(lodash.omit(soldier, 'duties'));
       expect(res.statusCode).toEqual(201);
-      expect(res.body).toEqual(testSoldiers[0]);
+      expect(res.body).toEqual(soldier);
     });
 
     test('Should return code 400 if missing a field', async () => {
       const res = await request(server)
         .post(`/soldiers`)
-        .send(lodash.omit(testSoldiers[0], ['duties', 'rank']));
+        .send(lodash.omit(createTestSoldier(0), ['duties', 'rank']));
       expect(res.statusCode).toEqual(400);
     });
 
     test('Should return code 400 if a field is the wrong type', async () => {
       const res = await request(server)
         .post(`/soldiers`)
-        .send({ ...testSoldiers[0], rank: 4 });
+        .send({ ...createTestSoldier(0), rank: 4 });
       expect(res.statusCode).toEqual(400);
     });
   });
 
   describe('GET /soldiers', () => {
+    const soldiers = Array.from({ length: 3 }, (_, i) => {
+      return createTestSoldier(i);
+    });
+
     beforeEach(async () => {
-      await collection.insertMany(testSoldiers.map((val) => ({ ...val })));
+      await collection.insertMany(JSON.parse(JSON.stringify(soldiers)));
```

Comment:
Why ?

Reply:
because insert is changing the object, so I want to copy it before passing to function

Reply:
resolved